src/styles/main.scss

```diff
@@ -0,0 +1,62 @@
+* {
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+}
+
+body {
+  display: flex;
+  justify-content: center;
+  height: 100vh;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  align-content: center;
+  max-width: 706px;
+  gap: 1px;
+
+  @for $number from 1 through 31 {
+    & :nth-child(#{$number})::before {
+      content: '#{$number}';
+      position: absolute;
```

Comment:
Move position absolute outside the loop, and actually as I see you have centered with display flex, so maybe you don't need to set position for these elements at all?